wetlab/views.py

```diff
@@ -2093,167 +2093,91 @@ def add_user_lot_commercial_kit(request):
-            for key, value in require_to_update.items():
-                if value == "newLibPreparation":
-                    if not libraryPreparation.objects.filter(sample_id__pk__exact=key):
-                        continue
-                    lib_prep_obj = libraryPreparation.objects.get(
-                        sample_id__pk__exact=key
-                    )
-                    # lib_prep_obj.set_state('Reused')
-                    lib_prep_obj.set_increase_reuse()
-                elif value == "newPool":
-                    pass
-                else:
-                    continue
-        """
+    # Get Samples model fields with its verbose name, and dropdown lists for each of them
+    fields_info = core.utils.samples.sample_table_fields(__package__)
 
-        return render(
-            request,
-            "wetlab/record_sample.html",
-            {"reprocess_result": sample_recorded["reprocess_result"]},
-        )
+    # Record new samples
+    if request.method == "POST" and request.POST["action"] == "record_samples":
+        # Get data from POST
+        req_user = request.user.username
+        excel_data = json.loads(request.POST["record_table_data"])
+        header = json.loads(request.POST["record_table_header"])
 
-    # display the form to show the samples in pre-defined state that user requested to complete
-    elif (
-        request.method == "POST"
-        and request.POST["action"] == "select_samples_pre_defined"
-    ):
-        if "samples_in_list" in request.POST:
-            pre_defined_samples_id = request.POST.getlist("samples")
-        sample_recorded = core.utils.samples.prepare_sample_project_input_table(
-            pre_defined_samples_id
+        # Change excel header (verbose_name) to field names
+        field_names = core.utils.samples.sheet_header_to_field_name(
+            header, fields_info["fields"]
         )
-        return render(
-            request,
-            "wetlab/record_sample.html",
-            {"sample_recorded": sample_recorded},
+        # Convert excel list-list to dictionary with field_names
+        excel_json_data = core.utils.common.jspreadsheet_to_dict(
+            field_names, excel_data
         )
-
-    # Add the additional information related to the project
-    elif request.method == "POST" and request.POST["action"] == "sampleprojectdata":
-        sample_recorded = core.utils.samples.analyze_input_sample_project_fields(
-            request.POST
+        # validate mandatory and redundant samples
+        validation = core.utils.samples.validate_sample_data(
+            excel_json_data, req_user, __package__
         )
 
-        if request.POST["pending_pre_defined"] != "":
-            sample_recorded.update(
-                core.utils.samples.prepare_sample_project_input_table(
-                    request.POST["pending_pre_defined"].split(",")
+        for val in validation:
+            if not val["Validate"]:
+                return render(
```

Comment:
I think is quite annoyed if many samples are not validated. Then user only see the first not validated sample, then after corrected it, the next  not validated sample  is showed. Then corrected and same with others. This behavior is quite frustrating. 
Suggestion to get notice to user, in once,  all not validated samples

Reply:
All the validation errors are in the `validation` dictionary, and a table with all samples that got error is going to be rendered in the html. That for only detetects if any has error to send output the validation errors, or if all went well output the record result.